File: Core CS/Algorithms/Warmup/Birthday Cake Candles/birthdayCakeCandles.kt
import java.io.*
import java.text.*
import java.util.*
import kotlin.collections.*

fun birthdayCakeCandles(candles: Array<Int>): Int {
    val mx = candles.max()
    return candles.filter{it == candles.max()}.size
}

fun main(args: Array<String>) {
    val candlesCount = readLine()!!.trim().toInt()

    val candles = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    val result = birthdayCakeCandles(candles)

    println(result)
}

File: Core CS/Algorithms/Warmup/Birthday Cake Candles/birthdayCakeCandles.scala
import java.io._
import java.text._
import java.util._
import scala.io._


object Result {

    def birthdayCakeCandles(candles: Array[Int]): Int = {
        val mx = candles.max
        return candles.filter(_ == mx).length
    }

}

object Solution {
    def main(args: Array[String]) {
        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val candlesCount = StdIn.readLine.trim.toInt

        val candles = StdIn.readLine.replaceAll("\\s+$", "").split(" ").map(_.trim.toInt)

        val result = Result.birthdayCakeCandles(candles)

        printWriter.println(result)

        printWriter.close()
    }
}
